azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADOAuth2AutoConfiguration.java

```diff
@@ -22,18 +30,36 @@
 @PropertySource(value = "classpath:serviceEndpoints.properties")
 @EnableConfigurationProperties({AADAuthenticationProperties.class, ServiceEndpointsProperties.class})
 public class AADOAuth2AutoConfiguration {
-    private AADAuthenticationProperties aadAuthProps;
-    private ServiceEndpointsProperties serviceEndpointsProps;
+
+    private final TelemetryProxy telemetryProxy;
+
+    private final AADAuthenticationProperties aadAuthProps;
+
+    private final ServiceEndpointsProperties serviceEndpointsProps;
 
     public AADOAuth2AutoConfiguration(AADAuthenticationProperties aadAuthProperties,
-                                      ServiceEndpointsProperties serviceEndpointsProps) {
+                                      ServiceEndpointsProperties serviceEndpointsProps,
+                                      TelemetryProxy telemetryProxy) {
         this.aadAuthProps = aadAuthProperties;
         this.serviceEndpointsProps = serviceEndpointsProps;
+        this.telemetryProxy = telemetryProxy;
+
+        trackCustomEvent();
     }
 
     @Bean
     @ConditionalOnProperty(prefix = "azure.activedirectory", value = "active-directory-groups")
     public OAuth2UserService<OidcUserRequest, OidcUser> oidcUserService() {
         return new AADOAuth2UserService(aadAuthProps, serviceEndpointsProps);
     }
+
+    private void trackCustomEvent() {
+        if (aadAuthProps.isAllowTelemetry()) {
+            final Map<String, String> events = new HashMap<>();
+
+            events.put(SERVICE_NAME, getClassPackageSimpleName(AADOAuth2AutoConfiguration.class));
+
+            telemetryProxy.trackEvent(ClassUtils.getUserClass(getClass()).getSimpleName(), events);
```

Comment:
Is the `getClassPackageSimpleName(AADOAuth2AutoConfiguration.class)` and `ClassUtils.getUserClass(getClass()).getSimpleName()` doing the same thing?

Reply:
No, one for class package simple name, and the other for class simple name